In [1]:
# Documentación libraría pandas
# https://pandas.pydata.org/docs/index.html
#importar libraría pandas
import pandas as pd

In [2]:
# dataset sobre consumo de alcohol en estudantes
# https://www.kaggle.com/uciml/student-alcohol-consumption
df = pd.read_csv('../datasets/estudantes_portugues.csv',sep=';')

In [3]:
# Abre o ficheiro directamente desde a rede (busca o enderezo no repo de github)
#url = ''
#df = pd.read_csv(url)

In [4]:
# mostrar o dataframe
df

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
644     MS   F   19       R     GT3       T     2     3  services     other   
645     MS   F   18       U     LE3       T     3     1   teacher  services   
646     MS   F   18       U     GT3       T     1     1     other     other   
647     MS   M   17       U     LE3       T     3     1  services  services   
648     MS   M   18       R     LE3       T     3     2  services     other   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        4   0  11  11  
1    ...      5        3      3     1     1      3        2   9  11  11  
2    ...      4        3      2     2     3      3        6  12  13  12  
3    ...      3        2      2     1     1      5        0  14  14  14  
4    ...      4        3      2     1     2      5        0  11  13  13  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
644  ...      5        4      2     1     2      5        4  10  11  10  
645  ...      4        3      4     1     1      1        4  15  15  16  
646  ...      1        1      1     1     1      5        6  11  12   9  
647  ...      2        4      5     3     4      2        6  10  10  10  
648  ...      4        4      1     3     4      5        4  10  11  11  

[649 rows x 33 columns]

In [5]:
# Mostra os estatísticos básicos do dataframe:
df.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  649.000000  649.000000  649.000000  649.000000  649.000000  649.000000   
mean    16.744222    2.514638    2.306626    1.568567    1.930663    0.221880   
std      1.218138    1.134552    1.099931    0.748660    0.829510    0.593235   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    1.000000    1.000000    1.000000    0.000000   
50%     17.000000    2.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  649.000000  649.000000  649.000000  649.000000  649.000000  649.000000   
mean     3.930663    3.180277    3.184900    1.502311    2.280431    3.536210   
std      0.955717    1.051093    1.175766    0.924834    1.284380    1.446259   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    2.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  649.000000  649.000000  649.000000  649.000000  
mean     3.659476   11.399076   11.570108   11.906009  
std      4.640759    2.745265    2.913639    3.230656  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000   10.000000   10.000000   10.000000  
50%      2.000000   11.000000   11.000000   12.000000  
75%      6.000000   13.000000   13.000000   14.000000  
max     32.000000   19.000000   19.000000   19.000000

#Busca nos estatísticos a seguinte información: (Ao final do Notebook podes ver o significado de cada campo):
#Media de idade
16.74
#Desviación típica na variable tempo libre
1.051
#Media do nivel de saúde
3.53
#Valor mínimo de tempo de estudo
1
#Valor máximo de faltas a clase
32

In [6]:
# mostra o tipo de datos de cada columna
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 649 entries, 0 to 648
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      649 non-null    object
 1   sex         649 non-null    object
 2   age         649 non-null    int64 
 3   address     649 non-null    object
 4   famsize     649 non-null    object
 5   Pstatus     649 non-null    object
 6   Medu        649 non-null    int64 
 7   Fedu        649 non-null    int64 
 8   Mjob        649 non-null    object
 9   Fjob        649 non-null    object
 10  reason      649 non-null    object
 11  guardian    649 non-null    object
 12  traveltime  649 non-null    int64 
 13  studytime   649 non-null    int64 
 14  failures    649 non-null    int64 
 15  schoolsup   649 non-null    object
 16  famsup      649 non-null    object
 17  paid        649 non-null    object
 18  activities  649 non-null    object
 19  nursery     649 non-null    object
 20  higher    

In [7]:
# Mostra as 3 primeiras  filas do dataframe
df.head(3)

school sex  age address famsize Pstatus  Medu  Fedu     Mjob     Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home  teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home    other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home    other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        4   0  11  11  
1      5        3      3     1     1      3        2   9  11  11  
2      4        3      2     2     3      3        6  12  13  12  

[3 rows x 33 columns]

In [8]:
# Mostra as columnas school, sex e age
df[['school','sex','age']]

school sex  age
0       GP   F   18
1       GP   F   17
2       GP   F   15
3       GP   F   15
4       GP   F   16
..     ...  ..  ...
644     MS   F   19
645     MS   F   18
646     MS   F   18
647     MS   M   17
648     MS   M   18

[649 rows x 3 columns]

In [9]:
# Mostra as 3 primeiras filas das columnas school, sex e age
df[['school','sex','age']].head(3)

school sex  age
0     GP   F   18
1     GP   F   17
2     GP   F   15

In [10]:
# Mostra 4 filas aleatorias das columnas school, sex e age
df[['school','sex','age']].sample(4)

school sex  age
542     MS   F   16
538     MS   F   16
373     GP   F   17
613     MS   F   18

In [11]:
# Mostra os nomes das columnas
df.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

In [12]:
# Mostra información sobre os índices
df.index

RangeIndex(start=0, stop=649, step=1)

In [13]:
# Mostra a columna de ausencias como unha serie
df.absences
#df['absences']

0      4
1      2
2      6
3      0
4      0
      ..
644    4
645    4
646    6
647    6
648    4
Name: absences, Length: 649, dtype: int64

In [14]:
# Mostra a columna de ausencias como un dataframe
df[['absences']]

absences
0           4
1           2
2           6
3           0
4           0
..        ...
644         4
645         4
646         6
647         6
648         4

[649 rows x 1 columns]

In [15]:
# crea unha variable s_ausencias coa serie das ausencias
s_ausencias = df.absences
# crea unha variable df_ausencias como dataframe de ausencias
df_ausencias = df[['absences']]

In [16]:
# mostra os 3 primeiros datos da serie s_ausencias
s_ausencias.head(3)

0    4
1    2
2    6
Name: absences, dtype: int64

In [17]:
# mostra as 7 primeiras filas do dataframe df_ausencias
df_ausencias.head(7)

absences
0         4
1         2
2         6
3         0
4         0
5         6
6         0

In [18]:
# Como ves, o tipo Series e o tipo Dataframe comparten algunhas funcións: .head(), .tail(), .sample()

In [19]:
# Mostra a terceira fila do dataframe orixinal  (como un dataframe)
df.iloc[[2]]

school sex  age address famsize Pstatus  Medu  Fedu     Mjob   Fjob  ...  \
2     GP   F   15       U     LE3       T     1     1  at_home  other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
2      4        3      2     2     3      3        6  12  13  12  

[1 rows x 33 columns]

In [20]:
# Mostra as filas 0, 2, 5 do dataframe orixinal
df.iloc[[0,2,5]]

school sex  age address famsize Pstatus  Medu  Fedu      Mjob     Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4   at_home  teacher  ...   
2     GP   F   15       U     LE3       T     1     1   at_home    other  ...   
5     GP   M   16       U     LE3       T     4     3  services    other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        4   0  11  11  
2      4        3      2     2     3      3        6  12  13  12  
5      5        4      2     1     2      5        6  12  12  13  

[3 rows x 33 columns]

In [21]:
# Escribe ao ficheiro estudantes1.csv os datos de sexo, idade, tempo libre, tempo de estudo e tempo de saír 
novo_df = df[['sex','age','freetime','studytime','goout']]
novo_df.to_csv('output/estudantes1.csv')

In [22]:
# Le o novo ficheiro creado e garda os datos no dataframe df_estudantes1
df_estudantes1 = pd.read_csv('output/estudantes1.csv',index_col=0)
df_estudantes1.head(3)
#novo_csv.sample(5)

sex  age  freetime  studytime  goout
0   F   18         3          2      4
1   F   17         3          2      3
2   F   15         3          2      2

In [23]:
# Escribe ao ficheiro estudantes2.csv os datos de idade, tempo libre, saúde, relación de parella e se queren optar a educación superior
# Nome do ficheiro: estudantes2.csv
# Utiliza o separador ;
# Non gardes as cabeceiras
# Non gardes os índices
novo_df = df[['age','freetime','health','romantic','higher']]
#novo_df.sample()
novo_df.to_csv('output/estudantes2.csv',sep=';',header=False,index=False)

In [24]:
# Le o novo ficheiro creado e garda os datos no dataframe df_estudantes2
# Utiliza os seguintes nomes para as columnas: idade, tempo libre, saude, emparellado, edu. superior
df_estudantes2 = pd.read_csv('output/estudantes2.csv', sep=';', header=None, names=['idade','tempo libre','saude','en parella','edu. superior'])
df_estudantes2.head(3)

idade  tempo libre  saude en parella edu. superior
0     18            3      3         no           yes
1     17            3      3         no           yes
2     15            3      3         no           yes

Attributes for both student-mat.csv (Math course) and student-por.csv (Portuguese language course) datasets:

    school - student's school (binary: 'GP' - Gabriel Pereira or 'MS' - Mousinho da Silveira)
    sex - student's sex (binary: 'F' - female or 'M' - male)
    age - student's age (numeric: from 15 to 22)
    address - student's home address type (binary: 'U' - urban or 'R' - rural)
    famsize - family size (binary: 'LE3' - less or equal to 3 or 'GT3' - greater than 3)
    Pstatus - parent's cohabitation status (binary: 'T' - living together or 'A' - apart)
    Medu - mother's education (numeric: 0 - none, 1 - primary education (4th grade), 2 – 5th to 9th grade, 3 – secondary education or 4 – higher education)
    Fedu - father's education (numeric: 0 - none, 1 - primary education (4th grade), 2 – 5th to 9th grade, 3 – secondary education or 4 – higher education)
    Mjob - mother's job (nominal: 'teacher', 'health' care related, civil 'services' (e.g. administrative or police), 'at_home' or 'other')
    Fjob - father's job (nominal: 'teacher', 'health' care related, civil 'services' (e.g. administrative or police), 'at_home' or 'other')
    reason - reason to choose this school (nominal: close to 'home', school 'reputation', 'course' preference or 'other')
    guardian - student's guardian (nominal: 'mother', 'father' or 'other')
    traveltime - home to school travel time (numeric: 1 - <15 min., 2 - 15 to 30 min., 3 - 30 min. to 1 hour, or 4 - >1 hour)
    studytime - weekly study time (numeric: 1 - <2 hours, 2 - 2 to 5 hours, 3 - 5 to 10 hours, or 4 - >10 hours)
    failures - number of past class failures (numeric: n if 1<=n<3, else 4)
    schoolsup - extra educational support (binary: yes or no)
    famsup - family educational support (binary: yes or no)
    paid - extra paid classes within the course subject (Math or Portuguese) (binary: yes or no)
    activities - extra-curricular activities (binary: yes or no)
    nursery - attended nursery school (binary: yes or no)
    higher - wants to take higher education (binary: yes or no)
    internet - Internet access at home (binary: yes or no)
    romantic - with a romantic relationship (binary: yes or no)
    famrel - quality of family relationships (numeric: from 1 - very bad to 5 - excellent)
    freetime - free time after school (numeric: from 1 - very low to 5 - very high)
    goout - going out with friends (numeric: from 1 - very low to 5 - very high)
    Dalc - workday alcohol consumption (numeric: from 1 - very low to 5 - very high)
    Walc - weekend alcohol consumption (numeric: from 1 - very low to 5 - very high)
    health - current health status (numeric: from 1 - very bad to 5 - very good)
    absences - number of school absences (numeric: from 0 to 93)

These grades are related with the course subject, Math or Portuguese:

    G1 - first period grade (numeric: from 0 to 20)
    G2 - second period grade (numeric: from 0 to 20)
    G3 - final grade (numeric: from 0 to 20, output target)
